### ISSUE TITLE
Safeguarding A Decorated Request Object

### ISSUE THREAD

**Issue by gordysc** [hapijs/hapi#3408]
When dealing with a HapiJS server w/ multiple connections, there are no friendly safeguards to prevent plugins from decorating the request object multiple times.  For example:

```
exports.register = (server, options, next) => {
  server.decorate('request', 'decoratedFunctionName', .... );
}
```

As stated in HapiJS's documentation:

> Note that decorations apply to the entire server and all its connections regardless of current selection.

Since we are registering the plugin across multiple connections, the only way to get the above to work is using a try/catch block.  It'd be super useful if there was a mechanism to help check for an existing decoration.  Perhaps something as simple as a method that'd return an array of existing decorations on the request object, like so:

`server.inspect('request')`
and return:
`['decoratedFunctionName', ...]`

Related issues:
https://github.com/hapijs/discuss/issues/415
https://github.com/hapijs/hapi-auth-cookie/issues/119

**Comment by AdriVanHoudt** [comment]
@gordysc you can access them via `server._decorations` but that is not a "public" api. Just exposing that would solve your case no?

**Comment by gordysc** [comment]
@AdriVanHoudt Yes, that's exactly what I'd love to have.  I imagine others would benefit from this being public as well.

**Comment by AdriVanHoudt** [comment]
I totally agree, if @hueniverse agrees a PR to make the decorations public is welcome!

**Comment by gordysc** [comment]
I'll gladly submit the PR if he's on board.

**Comment by AdriVanHoudt** [comment]
He might be more willing when the PR is there already 😉

**Comment by gordysc** [comment]
We can close this issue out and I'll track progress in the referenced PR.

**Comment by AdriVanHoudt** [comment]
@gordysc if you add `closes #3408` to your commit message github will auto link the commit and PR to this issue and when the PR gets merged the issue gets closed automatically :D

**Comment by gordysc** [comment]
@AdriVanHoudt good to know! You learn something new everyday :)

closes #3409
